### ISSUE TITLE
Error in installation

### ISSUE THREAD

**Issue by waitasecant** [fraymio/modis-tools#48]
The package gives error in installation.

`pip install modis-tools`

```
Collecting modis-tools
  Using cached modis_tools-1.1.3-py3-none-any.whl.metadata (10 kB)
Collecting requests==2.* (from modis-tools)
  Using cached requests-2.31.0-py3-none-any.whl.metadata (4.6 kB)
Collecting pydantic==1.* (from modis-tools)
  Using cached pydantic-1.10.15-py3-none-any.whl.metadata (150 kB)
Collecting python-dateutil==2.* (from modis-tools)
  Downloading python_dateutil-2.9.0.post0-py2.py3-none-any.whl.metadata (8.4 kB)
Collecting shapely==1.* (from modis-tools)
  Using cached Shapely-1.8.5.post1.tar.gz (200 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error

  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [33 lines of output]
      Traceback (most recent call last):
        File "D:\Saroja\MODIS\venv\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 353, in <module>
          main()
        File "D:\Saroja\MODIS\venv\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "D:\Saroja\MODIS\venv\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 112, in get_requires_for_build_wheel
          backend = _build_backend()
                    ^^^^^^^^^^^^^^^^
        File "D:\Saroja\MODIS\venv\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 77, in _build_backend
          obj = import_module(mod_path)
                ^^^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\Himanshu\AppData\Local\Programs\Python\Python312\Lib\importlib\__init__.py", line 90, in import_module
          return _bootstrap._gcd_import(name[level:], package, level)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
        File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
        File "<frozen importlib._bootstrap>", line 1310, in _find_and_load_unlocked
        File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
        File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
        File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
        File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
        File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
        File "<frozen importlib._bootstrap_external>", line 994, in exec_module
        File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
        File "C:\Users\Himanshu\AppData\Local\Temp\pip-build-env-ldusm2_z\overlay\Lib\site-packages\setuptools\__init__.py", line 16, in <module>
          import setuptools.version
        File "C:\Users\Himanshu\AppData\Local\Temp\pip-build-env-ldusm2_z\overlay\Lib\site-packages\setuptools\version.py", line 1, in <module>
          import pkg_resources
        File "C:\Users\Himanshu\AppData\Local\Temp\pip-build-env-ldusm2_z\overlay\Lib\site-packages\pkg_resources\__init__.py", line 2191, in <module>
          register_finder(pkgutil.ImpImporter, find_on_path)
                          ^^^^^^^^^^^^^^^^^^^
      AttributeError: module 'pkgutil' has no attribute 'ImpImporter'. Did you mean: 'zipimporter'?
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.


**Your Environment:**
- Python version- 3.12.1
- `modis_tools` version used: 1.1.3(latest)
-  pip list
Package Version
pip     24.0
```

**Any Additional context**
I first tried installing the package, which errored while installing. Then, I made a virtual environment and then tried installation, which again errored.

**Comment by jamie-sgro** [comment]
@waitasecant could you reply with the commands you used to create the venv and install. I tried on a fresh venv for both MacOS and Windows and I was unable to replicate. 

Could you run these two commands as well so we get a better idea of your environment?
 - Python version (`python -c "import sys; print(sys.version.replace('\n', ' '))"`)
 - Operation System Information (`python -c "import platform; print(platform.platform())"`)

<details>
<summary>Mac Install </summary>

```

(base) jamiesgro@C02F12LP0KPF-jamiesgro-Fraym test % python3 -m venv venv` 
bquote>  
(base) jamiesgro@C02F12LP0KPF-jamiesgro-Fraym test % python3 -m venv venv`
bquote> 
(base) jamiesgro@C02F12LP0KPF-jamiesgro-Fraym test % python3 -m venv venv 
(base) jamiesgro@C02F12LP0KPF-jamiesgro-Fraym test % source venv/bin/activate
(venv) (base) jamiesgro@C02F12LP0KPF-jamiesgro-Fraym test % pip freeze
(venv) (base) jamiesgro@C02F12LP0KPF-jamiesgro-Fraym test % pip install modis-tools
Collecting modis-tools
  Using cached modis_tools-1.1.3-py3-none-any.whl (26 kB)
Collecting requests==2.* (from modis-tools)
  Using cached requests-2.31.0-py3-none-any.whl (62 kB)
Collecting pydantic==1.* (from modis-tools)
  Using cached pydantic-1.10.15-cp311-cp311-macosx_11_0_arm64.whl (2.5 MB)
Collecting python-dateutil==2.* (from modis-tools)
  Using cached python_dateutil-2.9.0.post0-py2.py3-none-any.whl (229 kB)
Collecting shapely==1.* (from modis-tools)
  Using cached Shapely-1.8.5.post1-cp311-cp311-macosx_11_0_arm64.whl (1.1 MB)
Collecting tqdm>=4.42.0 (from modis-tools)
  Using cached tqdm-4.66.2-py3-none-any.whl (78 kB)
Collecting typing-extensions>=4.2.0 (from pydantic==1.*->modis-tools)
  Using cached typing_extensions-4.11.0-py3-none-any.whl (34 kB)
Collecting six>=1.5 (from python-dateutil==2.*->modis-tools)
  Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
Collecting charset-normalizer<4,>=2 (from requests==2.*->modis-tools)
  Using cached charset_normalizer-3.3.2-cp311-cp311-macosx_11_0_arm64.whl (118 kB)
Collecting idna<4,>=2.5 (from requests==2.*->modis-tools)
  Using cached idna-3.6-py3-none-any.whl (61 kB)
Collecting urllib3<3,>=1.21.1 (from requests==2.*->modis-tools)
  Using cached urllib3-2.2.1-py3-none-any.whl (121 kB)
Collecting certifi>=2017.4.17 (from requests==2.*->modis-tools)
  Using cached certifi-2024.2.2-py3-none-any.whl (163 kB)
Installing collected packages: urllib3, typing-extensions, tqdm, six, shapely, idna, charset-normalizer, certifi, requests, python-dateutil, pydantic, modis-tools
Successfully installed certifi-2024.2.2 charset-normalizer-3.3.2 idna-3.6 modis-tools-1.1.3 pydantic-1.10.15 python-dateutil-2.9.0.post0 requests-2.31.0 shapely-1.8.5.post1 six-1.16.0 tqdm-4.66.2 typing-extensions-4.11.0 urllib3-2.2.1

[notice] A new release of pip is available: 23.1.2 -> 24.0
[notice] To update, run: pip install --upgrade pip
(venv) (base) jamiesgro@C02F12LP0KPF-jamiesgro-Fraym test % pip freeze
certifi==2024.2.2
charset-normalizer==3.3.2
idna==3.6
modis-tools==1.1.3
pydantic==1.10.15
python-dateutil==2.9.0.post0
requests==2.31.0
Shapely==1.8.5.post1
six==1.16.0
tqdm==4.66.2
typing_extensions==4.11.0
urllib3==2.2.1
```

</details>


<details>
<summary>Windows Install</summary>

```
C:\...\test>python -m venv myenv

C:\...\test>myenv\Scripts\activate

(myenv) C:\...\test>pip freeze

(myenv) C:\...\test>pip install modis-tools
Collecting modis-tools
  Using cached modis_tools-1.1.3-py3-none-any.whl (26 kB)
Collecting shapely==1.*
  Using cached Shapely-1.8.5.post1-cp310-cp310-win_amd64.whl (1.3 MB)
Collecting requests==2.*
  Downloading requests-2.31.0-py3-none-any.whl (62 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 62.6/62.6 KB 3.5 MB/s eta 0:00:00
Collecting tqdm>=4.42.0
  Downloading tqdm-4.66.2-py3-none-any.whl (78 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 78.3/78.3 KB 4.3 MB/s eta 0:00:00
Collecting pydantic==1.*
  Using cached pydantic-1.10.15-cp310-cp310-win_amd64.whl (2.1 MB)
Collecting python-dateutil==2.*
  Downloading python_dateutil-2.9.0.post0-py2.py3-none-any.whl (229 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 229.9/229.9 KB 7.1 MB/s eta 0:00:00
Collecting typing-extensions>=4.2.0
  Using cached typing_extensions-4.11.0-py3-none-any.whl (34 kB)
Collecting six>=1.5
  Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
Collecting idna<4,>=2.5
  Downloading idna-3.6-py3-none-any.whl (61 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 61.6/61.6 KB ? eta 0:00:00
Collecting charset-normalizer<4,>=2
  Downloading charset_normalizer-3.3.2-cp310-cp310-win_amd64.whl (100 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100.3/100.3 KB 5.6 MB/s eta 0:00:00
Collecting certifi>=2017.4.17
  Downloading certifi-2024.2.2-py3-none-any.whl (163 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 163.8/163.8 KB 9.6 MB/s eta 0:00:00
Collecting urllib3<3,>=1.21.1
  Downloading urllib3-2.2.1-py3-none-any.whl (121 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 121.1/121.1 KB 6.9 MB/s eta 0:00:00
Collecting colorama
  Downloading colorama-0.4.6-py2.py3-none-any.whl (25 kB)
Installing collected packages: urllib3, typing-extensions, six, shapely, idna, colorama, charset-normalizer, certifi, tqdm, requests, python-dateutil, pydantic, modis-tools
Successfully installed certifi-2024.2.2 charset-normalizer-3.3.2 colorama-0.4.6 idna-3.6 modis-tools-1.1.3 pydantic-1.10.15 python-dateutil-2.9.0.post0 requests-2.31.0 shapely-1.8.5.post1 six-1.16.0 tqdm-4.66.2 typing-extensions-4.11.0 urllib3-2.2.1
WARNING: You are using pip version 22.0.4; however, version 24.0 is available.
You should consider upgrading via the 'C:\...test\myenv\Scripts\python.exe -m pip install --upgrade pip' command.
```

</details>

**Comment by waitasecant** [comment]
Thanks for your timely response. I hope this helps.

Command used to create the venv:
- `python -m venv venv`

Command to activate the venv: (In powershell)
- `venv\Scripts\Activate.ps1`

Results of the two commands:

(venv) PS D:\Saroja\MODIS> `python -c "import sys; print(sys.version.replace('\n', ' '))"`
3.12.1 (tags/v3.12.1:2305ca5, Dec  7 2023, 22:03:25) [MSC v.1937 64 bit (AMD64)]

(venv) PS D:\Saroja\MODIS> `python -c "import platform; print(platform.platform())"`
Windows-10-10.0.19045-SP0

**Comment by jamie-sgro** [comment]
Ah, I was able to recreate once I bumped my windows python version to 3.12. I'll create a task and have the team investigate. I don't have a time estimate for completion, but I appreciate you bringing it to my attention. Here's the full output:

```powershell
PS C:\Users\Jamie>  cd C:\Users\Jamie\Downloads\test
PS C:\Users\Jamie\Downloads\test> python -m venv venv
PS C:\Users\Jamie\Downloads\test> venv\Scripts\Activate.ps1
(venv) PS C:\Users\Jamie\Downloads\test> pip freeze
(venv) PS C:\Users\Jamie\Downloads\test> pip install modis-tools
Collecting modis-tools
  Obtaining dependency information for modis-tools from https://files.pythonhosted.org/packages/4b/b4/6021fab972d5254a117106abb76aeff17bedfb2a23f543bf7474ce0753ba/modis_tools-1.1.3-py3-none-any.whl.metadata
  Downloading modis_tools-1.1.3-py3-none-any.whl.metadata (10 kB)
Collecting requests==2.* (from modis-tools)
  Obtaining dependency information for requests==2.* from https://files.pythonhosted.org/packages/70/8e/0e2d847013cb52cd35b38c009bb167a1a26b2ce6cd6965bf26b47bc0bf44/requests-2.31.0-py3-none-any.whl.metadata
  Downloading requests-2.31.0-py3-none-any.whl.metadata (4.6 kB)
Collecting pydantic==1.* (from modis-tools)
  Obtaining dependency information for pydantic==1.* from https://files.pythonhosted.org/packages/27/17/a9872f20809e37ad03c523994ef3e0b7420c6508fe4553b7c0d8476ee03a/pydantic-1.10.15-py3-none-any.whl.metadata
  Downloading pydantic-1.10.15-py3-none-any.whl.metadata (150 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 150.6/150.6 kB 1.5 MB/s eta 0:00:00
Collecting python-dateutil==2.* (from modis-tools)
  Obtaining dependency information for python-dateutil==2.* from https://files.pythonhosted.org/packages/ec/57/56b9bcc3c9c6a792fcbaf139543cee77261f3651ca9da0c93f5c1221264b/python_dateutil-2.9.0.post0-py2.py3-none-any.whl.metadata
  Downloading python_dateutil-2.9.0.post0-py2.py3-none-any.whl.metadata (8.4 kB)
Collecting shapely==1.* (from modis-tools)
  Downloading Shapely-1.8.5.post1.tar.gz (200 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 200.9/200.9 kB 6.0 MB/s eta 0:00:00
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error

  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [33 lines of output]
      Traceback (most recent call last):
        File "C:\Users\Jamie\Downloads\test\venv\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 353, in <module>
          main()
        File "C:\Users\Jamie\Downloads\test\venv\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\Jamie\Downloads\test\venv\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 112, in get_requires_for_build_wheel
          backend = _build_backend()
                    ^^^^^^^^^^^^^^^^
        File "C:\Users\Jamie\Downloads\test\venv\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 77, in _build_backend
          obj = import_module(mod_path)
                ^^^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\Jamie\AppData\Local\Programs\Python\Python312\Lib\importlib\__init__.py", line 90, in import_module
          return _bootstrap._gcd_import(name[level:], package, level)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "<frozen importlib._bootstrap>", line 1381, in _gcd_import
        File "<frozen importlib._bootstrap>", line 1354, in _find_and_load
        File "<frozen importlib._bootstrap>", line 1304, in _find_and_load_unlocked
        File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
        File "<frozen importlib._bootstrap>", line 1381, in _gcd_import
        File "<frozen importlib._bootstrap>", line 1354, in _find_and_load
        File "<frozen importlib._bootstrap>", line 1325, in _find_and_load_unlocked
        File "<frozen importlib._bootstrap>", line 929, in _load_unlocked
        File "<frozen importlib._bootstrap_external>", line 994, in exec_module
        File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
        File "C:\Users\Jamie\AppData\Local\Temp\pip-build-env-ih3y7zn9\overlay\Lib\site-packages\setuptools\__init__.py", line 16, in <module>
          import setuptools.version
        File "C:\Users\Jamie\AppData\Local\Temp\pip-build-env-ih3y7zn9\overlay\Lib\site-packages\setuptools\version.py", line 1, in <module>
          import pkg_resources
        File "C:\Users\Jamie\AppData\Local\Temp\pip-build-env-ih3y7zn9\overlay\Lib\site-packages\pkg_resources\__init__.py", line 2191, in <module>
          register_finder(pkgutil.ImpImporter, find_on_path)
                          ^^^^^^^^^^^^^^^^^^^
      AttributeError: module 'pkgutil' has no attribute 'ImpImporter'. Did you mean: 'zipimporter'?
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.

[notice] A new release of pip is available: 23.2.1 -> 24.0
[notice] To update, run: python.exe -m pip install --upgrade pip
(venv) PS C:\Users\Jamie\Downloads\test> python -c "import sys; print(sys.version.replace('\n', ' '))"
3.12.0 (tags/v3.12.0:0fb18b0, Oct  2 2023, 13:03:39) [MSC v.1935 64 bit (AMD64)]
(venv) PS C:\Users\Jamie\Downloads\test> python -c "import platform; print(platform.platform())"
Windows-11-10.0.22621-SP0
(venv) PS C:\Users\Jamie\Downloads\test>

```

**Comment by waitasecant** [comment]
Hi @jamie-sgro

Thank you for being for the open source contributions. I've  made the necessary changes to address the problem. Additionally, I've submitted a pull request [#49](https://github.com/fraymio/modis-tools/pull/49) with the fix.

To resolve the issue,
- I basically added `setuptools` as a dependency in the `requirements.txt` file conditioning the python version >= 3.12. 
- The issue occurred because in Python 3.12 `setuptools` is no longer installed by default on virtual environment creation. (See https://docs.python.org/3/whatsnew/3.12.html and https://github.com/python/cpython/issues/95299)

From my end, now the package is getting installed without any error without breaking any change. Your feedback and contributions are greatly appreciated. If you have any further questions or concerns, please don't hesitate to let me know. I'm here to help!

Best regards,
Himanshu

**Comment by waitasecant** [comment]
@jamie-sgro
Reminder: I've made the required changes, please review the PR [#49](https://github.com/fraymio/modis-tools/pull/49)
Hope this helps :)

**Comment by jamie-sgro** [comment]
@waitasecant I've just redeployed PyPi and confirmed the same code that previously failed now succeeds on modis-tools version `1.1.4`

<details>
<summary>Log</summary>

```powershell
PS C:\Users\Jamie\Downloads\312> python -m venv venv
PS C:\Users\Jamie\Downloads\312> venv\Scripts\Activate.ps1
(venv) PS C:\Users\Jamie\Downloads\312> pip freeze
(venv) PS C:\Users\Jamie\Downloads\312> pip install modis-tools
Collecting modis-tools
  Obtaining dependency information for modis-tools from https://files.pythonhosted.org/packages/02/8b/d15967883183083b64dadf063d8d979eed1266d631c7c82e56d88b0c8fdb/modis_tools-1.1.4-py3-none-any.whl.metadata
  Downloading modis_tools-1.1.4-py3-none-any.whl.metadata (11 kB)
Collecting requests==2.* (from modis-tools)
  Obtaining dependency information for requests==2.* from https://files.pythonhosted.org/packages/70/8e/0e2d847013cb52cd35b38c009bb167a1a26b2ce6cd6965bf26b47bc0bf44/requests-2.31.0-py3-none-any.whl.metadata
  Using cached requests-2.31.0-py3-none-any.whl.metadata (4.6 kB)
Collecting pydantic==1.* (from modis-tools)
  Obtaining dependency information for pydantic==1.* from https://files.pythonhosted.org/packages/27/17/a9872f20809e37ad03c523994ef3e0b7420c6508fe4553b7c0d8476ee03a/pydantic-1.10.15-py3-none-any.whl.metadata
  Using cached pydantic-1.10.15-py3-none-any.whl.metadata (150 kB)
Collecting python-dateutil==2.* (from modis-tools)
  Obtaining dependency information for python-dateutil==2.* from https://files.pythonhosted.org/packages/ec/57/56b9bcc3c9c6a792fcbaf139543cee77261f3651ca9da0c93f5c1221264b/python_dateutil-2.9.0.post0-py2.py3-none-any.whl.metadata
  Using cached python_dateutil-2.9.0.post0-py2.py3-none-any.whl.metadata (8.4 kB)
Collecting shapely==2.* (from modis-tools)
  Obtaining dependency information for shapely==2.* from https://files.pythonhosted.org/packages/81/35/6d8b5cf9a747b94e8940619d914280fc4ce8e86e411ad19d39e7e8be036c/shapely-2.0.4-cp312-cp312-win_amd64.whl.metadata
  Downloading shapely-2.0.4-cp312-cp312-win_amd64.whl.metadata (7.2 kB)
Collecting tqdm>=4.42.0 (from modis-tools)
  Obtaining dependency information for tqdm>=4.42.0 from https://files.pythonhosted.org/packages/2a/14/e75e52d521442e2fcc9f1df3c5e456aead034203d4797867980de558ab34/tqdm-4.66.2-py3-none-any.whl.metadata
  Downloading tqdm-4.66.2-py3-none-any.whl.metadata (57 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 57.6/57.6 kB 1.0 MB/s eta 0:00:00
Collecting setuptools==69.2.0 (from modis-tools)
  Obtaining dependency information for setuptools==69.2.0 from https://files.pythonhosted.org/packages/92/e1/1c8bb3420105e70bdf357d57dd5567202b4ef8d27f810e98bb962d950834/setuptools-69.2.0-py3-none-any.whl.metadata
  Downloading setuptools-69.2.0-py3-none-any.whl.metadata (6.3 kB)
Collecting typing-extensions>=4.2.0 (from pydantic==1.*->modis-tools)
  Obtaining dependency information for typing-extensions>=4.2.0 from https://files.pythonhosted.org/packages/01/f3/936e209267d6ef7510322191003885de524fc48d1b43269810cd589ceaf5/typing_extensions-4.11.0-py3-none-any.whl.metadata
  Downloading typing_extensions-4.11.0-py3-none-any.whl.metadata (3.0 kB)
Collecting six>=1.5 (from python-dateutil==2.*->modis-tools)
  Obtaining dependency information for six>=1.5 from https://files.pythonhosted.org/packages/d9/5a/e7c31adbe875f2abbb91bd84cf2dc52d792b5a01506781dbcf25c91daf11/six-1.16.0-py2.py3-none-any.whl.metadata
  Downloading six-1.16.0-py2.py3-none-any.whl.metadata (1.8 kB)
Collecting charset-normalizer<4,>=2 (from requests==2.*->modis-tools)
  Obtaining dependency information for charset-normalizer<4,>=2 from https://files.pythonhosted.org/packages/b6/7c/8debebb4f90174074b827c63242c23851bdf00a532489fba57fef3416e40/charset_normalizer-3.3.2-cp312-cp312-win_amd64.whl.metadata
  Downloading charset_normalizer-3.3.2-cp312-cp312-win_amd64.whl.metadata (34 kB)
Collecting idna<4,>=2.5 (from requests==2.*->modis-tools)
  Obtaining dependency information for idna<4,>=2.5 from https://files.pythonhosted.org/packages/e5/3e/741d8c82801c347547f8a2a06aa57dbb1992be9e948df2ea0eda2c8b79e8/idna-3.7-py3-none-any.whl.metadata
  Downloading idna-3.7-py3-none-any.whl.metadata (9.9 kB)
Collecting urllib3<3,>=1.21.1 (from requests==2.*->modis-tools)
  Obtaining dependency information for urllib3<3,>=1.21.1 from https://files.pythonhosted.org/packages/a2/73/a68704750a7679d0b6d3ad7aa8d4da8e14e151ae82e6fee774e6e0d05ec8/urllib3-2.2.1-py3-none-any.whl.metadata
  Downloading urllib3-2.2.1-py3-none-any.whl.metadata (6.4 kB)
Collecting certifi>=2017.4.17 (from requests==2.*->modis-tools)
  Obtaining dependency information for certifi>=2017.4.17 from https://files.pythonhosted.org/packages/ba/06/a07f096c664aeb9f01624f858c3add0a4e913d6c96257acb4fce61e7de14/certifi-2024.2.2-py3-none-any.whl.metadata
  Downloading certifi-2024.2.2-py3-none-any.whl.metadata (2.2 kB)
Collecting numpy<3,>=1.14 (from shapely==2.*->modis-tools)
  Obtaining dependency information for numpy<3,>=1.14 from https://files.pythonhosted.org/packages/16/2e/86f24451c2d530c88daf997cb8d6ac622c1d40d19f5a031ed68a4b73a374/numpy-1.26.4-cp312-cp312-win_amd64.whl.metadata
  Downloading numpy-1.26.4-cp312-cp312-win_amd64.whl.metadata (61 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 61.0/61.0 kB ? eta 0:00:00
Collecting colorama (from tqdm>=4.42.0->modis-tools)
  Obtaining dependency information for colorama from https://files.pythonhosted.org/packages/d1/d6/3965ed04c63042e047cb6a3e6ed1a63a35087b6a609aa3a15ed8ac56c221/colorama-0.4.6-py2.py3-none-any.whl.metadata
  Downloading colorama-0.4.6-py2.py3-none-any.whl.metadata (17 kB)
Downloading modis_tools-1.1.4-py3-none-any.whl (26 kB)
Downloading pydantic-1.10.15-py3-none-any.whl (159 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 159.9/159.9 kB 4.8 MB/s eta 0:00:00
Using cached python_dateutil-2.9.0.post0-py2.py3-none-any.whl (229 kB)
Using cached requests-2.31.0-py3-none-any.whl (62 kB)
Using cached setuptools-69.2.0-py3-none-any.whl (821 kB)
Downloading shapely-2.0.4-cp312-cp312-win_amd64.whl (1.4 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.4/1.4 MB 9.2 MB/s eta 0:00:00
Using cached tqdm-4.66.2-py3-none-any.whl (78 kB)
Using cached certifi-2024.2.2-py3-none-any.whl (163 kB)
Downloading charset_normalizer-3.3.2-cp312-cp312-win_amd64.whl (100 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100.4/100.4 kB 6.0 MB/s eta 0:00:00
Downloading idna-3.7-py3-none-any.whl (66 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 66.8/66.8 kB 3.8 MB/s eta 0:00:00
Downloading numpy-1.26.4-cp312-cp312-win_amd64.whl (15.5 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 15.5/15.5 MB 8.8 MB/s eta 0:00:00
Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
Using cached typing_extensions-4.11.0-py3-none-any.whl (34 kB)
Using cached urllib3-2.2.1-py3-none-any.whl (121 kB)
Using cached colorama-0.4.6-py2.py3-none-any.whl (25 kB)
Installing collected packages: urllib3, typing-extensions, six, setuptools, numpy, idna, colorama, charset-normalizer, certifi, tqdm, shapely, requests, python-dateutil, pydantic, modis-tools
Successfully installed certifi-2024.2.2 charset-normalizer-3.3.2 colorama-0.4.6 idna-3.7 modis-tools-1.1.4 numpy-1.26.4 pydantic-1.10.15 python-dateutil-2.9.0.post0 requests-2.31.0 setuptools-69.2.0 shapely-2.0.4 six-1.16.0 tqdm-4.66.2 typing-extensions-4.11.0 urllib3-2.2.1

[notice] A new release of pip is available: 23.2.1 -> 24.0
[notice] To update, run: python.exe -m pip install --upgrade pip
(venv) PS C:\Users\Jamie\Downloads\312>
```

</details>

**Comment by waitasecant** [comment]
@jamie-sgro happy to help : )
